packages/apps/plugins/plugin-thread/src/ThreadPlugin.tsx

```diff
@@ -357,24 +357,32 @@ export const ThreadPlugin = (): PluginDefinition<ThreadPluginProvides> => {
               }
 
               if (data.subject instanceof DocumentType) {
-                // Sort threads by y-position.
-                // TODO(burdon): Should just use document position?
-                // TODO(zan): There's a bug here. When two threads have the same y-position they need to
-                // be sorted by their x-position.
-                const threads = data.subject.threads
+                const doc = data.subject;
+                const accessor = doc.content ? createDocAccessor(doc.content, ['content']) : undefined;
+
+                if (!accessor) {
+                  return null;
+                }
+
+                const getStartPosition = (cursor: string | undefined) => {
+                  const range = cursor ? getRangeFromCursor(accessor, cursor) : undefined;
+                  return range?.start ?? Number.MAX_SAFE_INTEGER;
+                };
+
+                const threads = doc.threads
                   .concat(state.staging[data.subject.id])
                   .filter(nonNullable)
-                  .toSorted((a, b) => state.threads[a.id] - state.threads[b.id]);
+                  .sort((a, b) => getStartPosition(a.anchor) - getStartPosition(b.anchor));
```

Comment:
sort vs toSorted? I suppose due to the concat/filter operations its not going to mutate the doc itself anyways

Reply:
Yeah `concat` and filter return new arrays so I'm not worried.